### PR TITLE
align comments based on prev/next line

### LINKED ISSUE #209: Comment alignment
I'm not sure if there's a way to do this already, but I'd like some comments to stay with the previous line's indentation. For example:

```
switch(foo) {
    case bar:
        baz();
        // falls through

    case biz:
        what();
}
```

Right now, ESFormatter converts it to:

```
switch(foo) {
    case bar:
        baz();
    // falls through

    case biz:
        what();
}
```

I think this could be generalized to say a comment that is not preceded with an empty line and is followed by an empty line should be indented to the previous code line's indent. (Maybe this could just be a new option to turn off or on?)

### LINKED ISSUE #270: Indent issue with comment in front of CatchClause
Expected:
```js
try {
  bla();
// comment
} catch(e) {
  throw e;
}
```
Actual:
```js
try {
  bla();
  // comment
} catch (e) {
  throw e;
}
```

### PR DESCRIPTION
**Pull request by millermedeiros**
this is still WIP but creating the pull request so you can see the new behavior. all the logic was implemented on [rocambole-indent](https://github.com/millermedeiros/rocambole-indent) and we are just reusing the `indent.alignComments()` method

still need to update the tests to match the new behavior but want to confirm if that is really the expected behavior... specially for cases like:

```js
try {
  foo();
} catch (e) {
// this won't be indented since it follows next line
);
```

fixes #209
fixes #270

### CODE REVIEW

File: test/compare/default/obj_expression-in.js [hunk 1]
```diff
@@ -85,8 +85,9 @@ Ext.define('VMS.model.User', {
     {
       name: 'LastName',
       type: 'string'
+      // align with previous line because of line break
 
-      // lorem ipsum
+      // align with "}"
```
Comment (jzaefferer) [inline test/compare/default/obj_expression-in.js]: If `align with "}"` applied, shouldn't this comment and the two below have less indent?

Comment (jzaefferer) [inline test/compare/default/obj_expression-in.js]: Nevermind, looking at input file...

**Comment by jzaefferer** [comment]
I've looked through all the updated tests. Looks good to me.

**Comment by mgutz** [comment]
That's an unusual style. I would have expected the comment above to be indented.

For example

```
function foo() {
// I would expect this comment to be indented
}
```

**Comment by millermedeiros** [comment]
I agree with @mgutz that it's weird and the rule that he described should be easy to implement and makes way more sense.

**Comment by millermedeiros** [comment]
uhm, there are a few edge cases like:

```js
function foo() {
// this comment should be indented

  return bar();
}
```

I considered that it would add the indent only if it was inside an empty block when in fact it should add if comment is first non-empty token after `{` or `[` or `(`.

will need to release yet another version of rocambole-indent :/

**Comment by millermedeiros** [comment]
I updated the `rocamboleIndent.alignComments()` method to handle all the edge cases that I could think of and decided to add a new option `indent.AlignComments` to toggle the behavior:

```js
// indent.AlignComments = true (default)
if (foo) {
  bar();
// aligned with "else" (new behavior)
} else {
  baz();
}
```

and:

```js
// indent.AlignComments = false
if (foo) {
  bar();
  // aligned with "bar();" (old behavior)
} else {
  baz();
}
```

I also added a setting `indent.CommentInsideEmptyBlock` that user can use to toggle between:

```js
// indent.CommentInsideEmptyBlock = 1 (default)
function noop() {
  // this will be indented
}
```

and:

```js
// indent.CommentInsideEmptyBlock = 0
function noop() {
// this won't be indented
}
```

**Comment by jzaefferer** [comment]
This looks good to me. Hopefully this is the last time we need to update line comment handling!

**Comment by millermedeiros** [comment]
squased commits and landed on master \o/

**Pull request closed without merging**